### PR TITLE
prefer webshot2 to webshot and solve #1858

### PR DESCRIPTION
**Pull request by atusy**
With this PR, html_screenshot` tries to use **webshot2** and fallbacks to **webshot**.

Also, user may prioritize **webshot** by `opts_chunk$set(webshot = "webshot")`.
Another possibility is to use `opts_knit$set(webshot = "webshot")`.
I'm fine with both ways, but choose the `opts_chunk` because it has controlls on each chunks.

### CODE REVIEW

File: R/plot.R [hunk 1]
```diff
@@ -488,18 +488,21 @@ need_screenshot = function(x, ...) {
...
 }
 
 webshot_available = local({
-  res = NULL  # cache the availablity of webshot/PhantomJS
+  res = NULL  # cache the availablity of webshot2 and webshot/PhantomJS
   function() {
     if (is.null(res))
-      res <<- loadable('webshot') && !is.null(getFromNamespace('find_phantom', 'webshot')())
+      res <<- c(
+        webshot2 = loadable('webshot2'),
```
Comment (yihui) [inline R/plot.R:503]: Should we test the availability of Chrome/Chromium?

**Review by yihui — APPROVED** [review]
Looks good to me. Just one small question. Thank you!

File: R/plot.R (hunk 1 above)
Comment (atusy) [reply R/plot.R:503]: I agree. Currently, **chromote** does not export `find_chrome`, so I'll open an issue there.
https://github.com/rstudio/chromote/blob/3a5329220d84de5f3a442cf05057d1f4ed7d3a7a/R/chrome.R#L17

Comment (atusy) [reply R/plot.R:503]: Oops, we do not need `chromote:::find_chrome` be exported as we use `getFromNamespace`

Comment (atusy) [reply R/plot.R]: Done via [`01d7c28` (#1918)](https://github.com/yihui/knitr/pull/1918/commits/01d7c28dd0073be1e97c7ca011c987ed743e089e)

**Review by yihui — APPROVED** [review]
Great. I'll carry on from here and merge in a few minutes. Thank you!

File: R/plot.R (hunk 1 above)
Comment (cderv) [reply R/plot.R]: @atusy I think it is still a good idea to suggest they export this function to allow test of chrome existence in other package. Worth a try ? 
Or suggest they add a `is_chrome_available()` in **webshot2**  ? 🤷

**Comment by yihui** [comment]
I forgot to mention that another possible idea is to use `pagedown::chrome_print()`. We have exported `pagedown::find_chrome()` there, and it is a CRAN package. I guess the only problem is the compatibility of arguments.

File: R/plot.R (hunk 1 above)
Comment (atusy) [reply R/plot.R]: @cderv I agree. It is better to have the function in **webshot2**. Then we can incorporate Yihui's refactoring https://github.com/yihui/knitr/commit/997398230c976bd10d24423fc84268a0a9be88a5#. I'll open issue tomorrow.

**Comment by cderv** [comment]
The function are a bit different in their way to find chrome, and specifically with the env var mechanism which is not the same. They both will find chrome but their could be a mismatch of version if you have several locally, as **webshot2** will use the one that **chromote** will use. 

Also `webshot:::find_phantom()` was unexported - does feel bad to do the same. 
I would stick to the one in **chromote** for now, and maybe ask them to put one in **webshot2** as a helper to test availability of chrome for other package ? or better that they do the check themselves maybe.

**Comment by cderv** [comment]
`find_chrome` is now exported from **chromote**. Not yet from **webshot2** I believe.

**Pull request merged**